### DISCUSSION TITLE
How to limit response fields for relationships?

### DISCUSSION THREAD

**Discussion by NaumenkoSergiy** [okuramasafumi/alba#326] [Q&A] [answered]
I'm interesting how to return all data, but limit some fields for specific relationships dynamically like this:

```
App::AgreementSerializer.new(
  @agreements,
  meta: pagy_pagination_metadata,
  fields: {
    organization: [:id, :slug, :name]
  }
)
```

**Comment by okuramasafumi** [comment]
@NaumenkoSergiy Could you provide the expected input and output?

**Comment by NaumenkoSergiy** [comment]
@okuramasafumi 

For example: `Agreement` has one `Organization, and `Organization` serializer has next attributes: `id`, `slug`, `name`, `logo`, `description`. But I want to have in response only specific fields from organization: 

```
App::AgreementSerializer.new(
  @agreements,
  meta: pagy_pagination_metadata,
  fields: {
    organization: [:id, :slug, :name]
  }
)
```
=>
```
{
  id: 1,
  name: 'test',
  desciption: 'description',
  organization: {
    id: 1,
    slug: 'test',
    name: Test
  }
}
```

**Reply by okuramasafumi** [reply]
What is in your `App::AgreementSerializer`? It should return what's expected with the following code:

```ruby
class App::AgreementSerializer
  include Alba::Resource

  attributes :id, :name, :description

  one :organization do
    attributes :id, :slug, :name
  end
end
```

**Comment by NaumenkoSergiy** [comment]
@okuramasafumi let's imagine that I have serializer what you provided. And sometimes I want to return only `id` for organization, or `id` and `slug`, not all attributes

**Reply by okuramasafumi** [reply, marked as answer]
I just wrote a test case for your use case.
https://github.com/okuramasafumi/alba/blob/main/test/usecases/field_selection_test.rb

Basically what we do here is to add another custom DSL to Alba. You can try this out and see how it goes. Let me know if you have some trouble.
